src/wp-includes/class-wp-query.php

```diff
@@ -3266,10 +3266,15 @@ public function get_posts() {
 			if ( ! empty( $sticky_posts ) ) {
 				$stickies = get_posts(
 					array(
-						'post__in'    => $sticky_posts,
-						'post_type'   => $post_type,
-						'post_status' => 'publish',
-						'nopaging'    => true,
+						'cache_results'          => $q['cache_results'],
+						'lazy_load_term_meta'    => $q['lazy_load_term_meta'],
+						'post__in'               => $sticky_posts,
+						'post_type'              => $post_type,
+						'post_status'            => 'publish',
+						'posts_per_page'         => count( $sticky_posts ),
```

Comment:
This is a minor back-compat break.

In the past, excluding this limited sticky posts to a maximum of five. With this change, the number of stickies is unlimited. 

I suspect this is a bug rather than a feature, as the [sticky posts support docs](https://developer.wordpress.org/themes/functionality/sticky-posts/) includes a code example for getting the first five stickies only.

If you do want to switch to all, you can change `post__in` to include and `get_posts()` will calculate `posts_per_page` for you. You may not wish to do this.

Reply:
I think we should remove this. I was trying to be clever and improve performance of the query with a limit. But this change should be happen in another PR. 

Oh dear, sticky posts are a mess.

Reply:
New ticket created - https://core.trac.wordpress.org/ticket/55444